### ISSUE TITLE
Continuous Ranked Probability Score (CRPS) metric

### ISSUE THREAD

**Issue by tmachnitzki** [PyTorchLightning/pytorch-lightning#7341]
## 🚀 Feature

The Continuous Ranked Probability Score (CRPS) is a metric for ensemble verification of continuous variables.
It is the continuous version of the  Ranked Probability Score (RPS), which is used for ensemble classification.

A nice description can be found here: https://www.xces.dkrz.de/about/problems/

### Motivation

Sampling from a GAN can be compared to having an ensemble of models. All existing metrics in Pytorch and Pytorch-Lightning do not allow for evaluating the quality of such multiple draws from a GAN, only evaluating each draw individually. 

Use cases would be where any generative model is used to predict a conditioned continuous field or even a single variable where all draws are strongly correlated, for example forecasting weather-patterns or disaster-evaluation. 
It will not be useful in the cases where the draws strongly uncorrelated, like in image-generation of faces and such.

### Pitch

I already have code for the CRPS written in pytorch, which I could provide including some tests.
Before opening a PR I just wanted to check if something like this score is really wanted.

### Alternatives

One could as well also provide the CRP for classification.
The CRPS can also be transformed into a skill score. A skill score would need as additional input a baseline to compare the skill against. This could for example be the condition in the case of a conditional GAN. (I have code for this as well)

### Additional context

For training GANs this score can be used as a loss function since it is 0 for a perfect ensemble and grows for worse ensembles (I have not tried that in practice yet).
